code/noRAM/app/src/main/java/com/example/noram/AttendeeActivity.java

```diff
@@ -83,39 +84,58 @@ public void onClick(View v) {
-                } else if (itemID == NAV_EVENTS) {
-                    selectedFragment = eventsFragment;
-                    headerString = getString(R.string.attendee_events_title);
-                } else if (itemID == NAV_PROFILE) {
-                    selectedFragment = profileFragment;
-                    headerString = getString(R.string.attendee_profile_title);
-                }
-
-                if (selectedFragment == null) {
-                    return false;
-                } else {
-                    // update the header text
-                    headerText.setText(headerString);
-                    // update the fragment container to show the selected fragment.
-                    fragmentManager.beginTransaction()
-                            .hide(activeFragment)
-                            .show(selectedFragment)
-                            .commit();
-                    fragmentManager.beginTransaction()
-                            .setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out)
-                            .replace(R.id.fragment_container_view, selectedFragment)
-                            .commit();
-                    return true;
-                }
+                return navigateTo(item);
             }
         });
     }
+
+    /**
+     * Navigate to the corresponding menu item
+     * @param item menuItem to navigate to
+     * @return true if navigation succeeded, else false
+     */
+    private boolean navigateTo(MenuItem item) {
+        Fragment selectedFragment = null;
+        String headerString = "";
+
+        // set the selectedFragment to the appropriate fragment
+        int itemID = item.getItemId();
+        if (itemID == NAV_SCAN) {
+            selectedFragment = qrFragment;
+            headerString = getString(R.string.scan_qr_code_title);
+        } else if (itemID == NAV_EVENTS) {
+            selectedFragment = eventsFragment;
+            headerString = getString(R.string.attendee_events_title);
+        } else if (itemID == NAV_PROFILE) {
+            selectedFragment = profileFragment;
+            headerString = getString(R.string.attendee_profile_title);
+        }
+
+        if (selectedFragment == null) {
+            return false;
+        } else {
+            // update the header text
+            headerText.setText(headerString);
+            // update the fragment container to show the selected fragment.
+            fragmentManager.beginTransaction()
+                    .hide(activeFragment)
+                    .setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out)
+                    .show(selectedFragment)
+                    .commitNow();
+            activeFragment = selectedFragment;
+            return true;
+        }
+    }
+
+    /**
+     * Go to the event passed by event by navigating to it
+     * @param event event to go to
+     */
+    @Override
+    public void goToEvent(Event event) {

```

Comment:
This works fine, but it might be more convenient (or efficient) to add a more general method that directly goes to the eventInfo page (it only needs to be passed the event ID).

Reply:
That's an interesting idea, I wasn't sure if there was a way to get the other fragment from the fragment though, which was why I went through the activity.